### PR TITLE
Extend .subs() to take more than one equation.

### PR DESCRIPTION
**Pull request by schymans**
This simplifies substitutions of equations:
` eq_Le.subs(eq_Dva, eq_alphaa)`
instead of
`eq_Le.subs({D_va: eq_Dva.rhs, alpha_a: eq_alphaa.rhs})`

### CODE REVIEW

File: essm/equations/_core.py [hunk 1]
```diff
@@ -159,19 +159,31 @@ def subs(self, *args, **kwargs):  # should mirror sympy.core.basic.subs
...
 
-        >>> from essm.equations.physics.thermodynamics import eq_Pa, eq_PN2_PO2
-        >>> from essm.variables.physics.thermodynamics import P_N2, P_O2
-        >>> eq_Pa.subs({P_N2: P_O2})
-        Eq(P_a, 2*P_O2 + P_wa)
-        >>> eq_Pa.subs(eq_PN2_PO2)
-        Eq(P_a, P_O2*x_N2/x_O2 + P_O2 + P_wa)
-
+        >>> from essm.equations.physics.thermodynamics import eq_Le,\
+            eq_Dva, eq_alphaa
```
Comment (jirikuncar) [inline essm/equations/_core.py]: have you tried with
```suggestion
        ... eq_Dva, eq_alphaa
```

File: essm/equations/_core.py [hunk 2]
```diff
@@ -159,19 +159,31 @@ def subs(self, *args, **kwargs):  # should mirror sympy.core.basic.subs
...
+        >>> from essm.variables.physics.thermodynamics import Le, D_va,\
+            alpha_a, T_a
```
Comment (jirikuncar) [inline essm/equations/_core.py]:
```suggestion
        ... alpha_a, T_a
```

File: essm/equations/_core.py [hunk 3]
```diff
@@ -155,23 +155,35 @@ def __add__(self, other):
...
+        Eq(Le, (T_a*p_alpha1 - p_alpha2)/(T_a*p_Dva1 - p_Dva2))
+        >>> eq_Le.subs(eq_Dva, eq_alphaa)
+        Eq(Le, (T_a*p_alpha1 - p_alpha2)/(T_a*p_Dva1 - p_Dva2))
         """
         if len(args) == 1:
             if isinstance(args[0], Eq):
                 arg1 = {args[0].lhs: args[0].rhs}
                 return Eq(self.lhs.subs(arg1, **kwargs),
                           self.rhs.subs(arg1, **kwargs))
+        else:
```
Comment (jirikuncar) [inline essm/equations/_core.py]: Using `else:` with `return` will make line 187 unreachable.

a) use `elif`
b) remove `else`, remove one level of indentation and remove 187-189.

Comment (schymans) [reply essm/equations/_core.py]: Line 187 should be reached if *args is a dictionary. Then, len(args) = 1, but it is not an equation, so the else statement will be omitted.

Comment (schymans) [reply essm/equations/_core.py]: Just to clarify, the original `.subs()` takes either a dictionary or a list of tuples or two items, the first of which will be substituted by the second in the expression. With the new definition of `.subs()` it is possible to provide 1, 2 or many equations instead.

Comment (schymans) [reply essm/equations/_core.py]: Dammit, it still only takes up to two equations. Haven't tested for this. Do you know how to extend it to many equations? I will provide an example.

Comment (schymans) [reply essm/equations/_core.py]: Just checked again, and this works: 
```python
from essm.equations.leaf.energy_water import eq_Rs_enbal, eq_El, eq_Hl, eq_Rll
eq_Rs_enbal.subs(eq_El, eq_Hl, eq_Rll)
```
returns:
> Eq(R_s, E_lmol*M_w*lambda_E + a_sh*epsilon_l*sigm*(T_l**4 - T_w**4) + a_sh*h_c*(-T_a + T_l))

I can't put it into the docstring, as the line is too long and if I add a line breat, the test fails.

Comment (jirikuncar) [reply essm/equations/_core.py]: So probably you want to use `elif`
```suggestion
        elif not kwargs:
```

run your current code with following cases:
```
args = (eq, ), kwargs = {} -> if
args = (), kwargs = {} -> else
args = (), kwargs = {foo: bar} -> else
```

**Comment by jirikuncar** [comment]
@schymans I have simplified the code a bit and joined all doctests.

**Review by jirikuncar — APPROVED** [review]

**Pull request merged**